### PR TITLE
Remove deprecated classes/methods usage from Legacy/Tests/VPUPlugin

### PR DESCRIPTION
**Pull request by GlebKazantaev**
This PR includes changes related to deprecated methods clean-up in the following components:
* Legacy API
* Functional Tests
* VPU Plugin

### CODE REVIEW

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 1]
```diff
@@ -195,14 +185,23 @@ ie::ICNNNetwork::Ptr FrontEnd::convertNetwork(ie::ICNNNetwork& network) {
...
+    pass_config->disable<ngraph::pass::ConvertGELU>();
+    pass_config->disable<ngraph::pass::SoftPlusDecomposition>();
+    pass_config->disable<ngraph::pass::ConvertMinimum>();
+    pass_config->disable<ngraph::pass::HSwishDecomposition>();
+    pass_config->disable<ngraph::pass::ConvertMatMulToFC>();
+    pass_config->disable<ngraph::pass::ConvertStridedSliceToCropMatcher>();
 
-    vpu::MergeSubsequentDSROperations().run_on_function(nGraphFunc);
+    const auto transformationsPredicate = [](const std::shared_ptr<const ngraph::Node>& node) -> bool {
+        return !!std::dynamic_pointer_cast<const ngraph::vpu::op::DynamicShapeResolver>(node->input_value(0).get_node_shared_ptr());
```
Comment (GlebKazantaev) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:200]: @gladilov-gleb @AndrewBakalinIntel do you remember the transformation that uses this callback? Or it applies for every transformation?

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:200]: The idea to disable transformations in dynamic case (MatMul\StridedSlice related passes)

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp [hunk 2]
```diff
@@ -3,11 +3,295 @@
...
+    ConvertSub() {
+        auto m = std::make_shared<ngraph::pattern::Matcher>(ngraph::pattern::wrap_type<ngraph::opset1::Subtract>());
+        register_matcher(m, get_callback<ngraph::opset1::Subtract>());
+    }
+};
+
+class ConvertMul: public ngraph::pass::MatcherPass {
+public:
+    ConvertMul() {
+        auto m = std::make_shared<ngraph::pattern::Matcher>(ngraph::pattern::wrap_type<ngraph::opset1::Multiply>());
```
Comment (ilyachur) [inline inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp:288]: Please add CC macros. You can find examples in the PR https://github.com/openvinotoolkit/openvino/pull/3795

Comment (GlebKazantaev) [reply inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp:288]: @ilyachur looks like we don't have `itt.hpp` with `MATCHER_PASS_SCOPE` macro for legacy_library which I think also need to be updated with itt. Could you handle it in a separate PR?

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp (hunk 1 above)
Comment (GlebKazantaev) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:198]: Callback changed to be set for particular passes.

File: inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp (hunk 2 above)
Comment (ilyachur) [reply inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp:288]: Oh yes, I thought it a transformation library. 
Yes we can handle it separately

**Review by ilyachur — APPROVED** [review]

**Comment by GlebKazantaev** [comment]
Waiting for internal VPU validation.

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 3]
```diff
@@ -195,14 +185,22 @@ ie::ICNNNetwork::Ptr FrontEnd::convertNetwork(ie::ICNNNetwork& network) {
...
+    auto pass_config = manager.get_pass_config();
+    pass_config->disable<ngraph::pass::ConvertGatherToGatherIEMatcher>();
+    pass_config->disable<ngraph::pass::ConvertGELU>();
+    pass_config->disable<ngraph::pass::SoftPlusDecomposition>();
+    pass_config->disable<ngraph::pass::ConvertMinimum>();
+    pass_config->disable<ngraph::pass::HSwishDecomposition>();
 
-    vpu::MergeSubsequentDSROperations().run_on_function(nGraphFunc);
+    auto transformationPredicate = [](const std::shared_ptr<const ngraph::Node>& node) -> bool {
+        return !!std::dynamic_pointer_cast<const ngraph::vpu::op::DynamicShapeResolver>(node->input_value(0).get_node_shared_ptr());
```
Comment (AndrewBakalinIntel) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:198]: Does `!!` makes sense?

Comment (GlebKazantaev) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:198]: Yes as we need to make bool from shared_ptr

**Pull request merged**